File: LeetCode/55.Jump_Game/55.Jump_Game.py
#!/usr/bin/env python3


class Solution:
    def canJump(self, nums: List[int]) -> bool:
        # greedy
        n = len(nums)
        max_reach = 0

        for i, step in enumerate(nums):
            if i > max_reach:
                return False

            max_reach = max(i + step, max_reach)
            if max_reach >= n - 1:
                return True

        return False

File: LeetCode/55.Jump_Game/55.Jump_Game.swift
class Solution {
    func canJump(_ nums: [Int]) -> Bool {
        let n = nums.count
        guard n > 1 else { return true }

        var reach = 0
        for i in 0 ..< n {
            if i > reach { return false }
            reach = max(reach, i + nums[i])
        }

        return true
    }
}
